service/frontend/admin_handler.go

```diff
@@ -2033,3 +2034,72 @@ func (adh *AdminHandler) StreamWorkflowReplicationMessages(
 	<-shutdownChan.Channel()
 	return nil
 }
+
+func (adh *AdminHandler) GetNamespace(ctx context.Context, request *adminservice.GetNamespaceRequest) (*adminservice.GetNamespaceResponse, error) {
+	req := &persistence.GetNamespaceRequest{
+		Name: request.GetNamespace(),
+		ID:   request.GetId(),
+	}
```

Comment:
input validation: one and only one attr is set?

Reply:
and Q: for replication, only namespace ID will be used?

Reply:
> input validation 

added. Since the **oneof** is used at request, there is no way to set both name and id at same time.

> for replication, only namespace ID will be used?

Currently seems only namespace ID is used, Add name to this API is to give some extendibility in case in the future we need to retrieve by name